airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -151,85 +153,150 @@ def print_job_logs(
         job_failed: bool = False,
         next_token: Optional[str] = None,
     ) -> Optional[str]:
-        """Prints the batch of logs to the Airflow task log and returns nextToken."""
-        log_client = boto3.client('logs')
-        response = {}
-
-        filter_pattern = FAILURE_LOG_FILTER if job_failed else DEFAULT_LOG_FILTER
+        """Prints the batch of Glue cloudwatch logs to the Airflow task log and returns nextToken."""
+        credentials = self.get_credentials(region_name=self.conn_region_name)
+        log_client = boto3.client(
+            'logs',
+            region_name=self.conn_region_name,
+            aws_access_key_id=credentials.access_key,
+            aws_secret_access_key=credentials.secret_key,
+        )
```

Comment:
Might be rather than directly use `boto3` use AwsBaseHook.get_session method instead, so it could be replaced by

```python
log_client = self.get_session(region_name=self.conn_region_name).client('logs')
```

Reply:
BTW, in your case you do not provide `aws_session_token` into the client. If user follow aws best practices and use temporary credentials obtain by role_arn, EC2 Instance IAM role, ECS Task IAM Role, IRSA EKS and etc. then token should be provided otherwise error will raised.

I thought token not required only for permanent credentials

Reply:
> 

I tried that but the issue is, in Glue connector, we are already using one client for Glue from AWSBaseHook object, so for cloudwatch I really do not want to create one more object of AWSBaseHook t get the boto3 client that's a reason I have added it directly through boto3. Correct

Reply:
In the existing Glue hook, we are using boto3 directly for log_client. I guess the reason for that will be a conflict of kwargs['client_type"]. since we are already using a glue client from AWSBaseHook in the same code, creating a new client for logs from the same AWSBaseHook class is not possible. I agree with your point that it might cause issue with role-based access.. let me check how we can solve it. I guess I should declare AWSLogsHook object and get a session from that only. I will try that and push new changes soon. Meanwhile, if you have any other workaround please feel free to provide the same.

Reply:
> creating a new client for logs from the same AWSBaseHook class is not possible

Yes you can not create different client type by call `AWSBaseHook.get_client_type` since latest changes.

https://github.com/apache/airflow/blob/3b6176929ccc0351554e5331d920c4ec65b33a38/airflow/providers/amazon/aws/hooks/base_aws.py#L438-L454

But you still can create `boto.session.Session()` and can use this session for create whatever client you wanted

https://github.com/apache/airflow/blob/3b6176929ccc0351554e5331d920c4ec65b33a38/airflow/providers/amazon/aws/hooks/base_aws.py#L432-L436

**Generic approach for boto3 client**
```python
session = boto3.session.Session(...)
client = session("awesome-client", ...)
```

**How it could be archived by `AwsGenericHook` / `AWSBaseHook` and any their child**
```python
hook = GlueJobHook(aws_conn_id="awesome_aws_conn_id", ...)
session = self.get_session(...)
client = session("awesome-client", ...)
```

Reply:
I have taken care of Log client from AWSBaseHook. It is working fine now. Please review now and let me know any further changes require.